gravitee-apim-gateway/gravitee-apim-gateway-security/gravitee-apim-gateway-security-jwt/src/main/java/io/gravitee/gateway/security/jwt/policy/CheckSubscriptionPolicy.java

```diff
@@ -48,10 +49,18 @@ public void execute(PolicyChain policyChain, ExecutionContext executionContext)
         final String api = (String) executionContext.getAttribute(ExecutionContext.ATTR_API);
         final String clientId = (String) executionContext.getAttribute(CONTEXT_ATTRIBUTE_CLIENT_ID);
         final String plan = (String) executionContext.getAttribute(ExecutionContext.ATTR_PLAN);
+        final String repositoryType = (String)  executionContext.getAttribute(CONTEXT_REPOSITORY_TYPE);
 
         executionContext.request().metrics().setSecurityType(JWT);
         executionContext.request().metrics().setSecurityToken(clientId);
 
+        // DB Less mode
+        if ("none".equals(repositoryType)) {
```

Comment:
Shouldn't we avoid having such hard coded behavior depending on the repository inside a Subscription process?

Reply:
I made some change a moved the logic to the subscription service

Reply:
@guillaumelamirand Are you OK with this new changes?